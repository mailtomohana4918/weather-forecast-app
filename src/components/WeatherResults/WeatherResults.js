import { Typography, Box, Card, CardMedia, CardContent } from '@material-ui/core'
import { makeStyles } from "@material-ui/core/styles";
import React from "react"
import weather from '../../weather.jpg'

const useStyles = makeStyles(() => ({
    boxContainer: {
        paddingTop: "24px",
        width: "600px",
        paddingLeft: "18px",
        paddingRight: "18px",      
    }
}));

//component renders for a successful response and displays weather info
export const WeatherResults = (props) => {
    const classes = useStyles();
    return (

        <Box className={classes.boxContainer}>
            <Card sx={{ maxWidth: 500}}>
                <CardMedia
                    component="img"
                    height="140"
                    image={weather}
                    alt="weather"
                />
                <CardContent >
                    <Typography gutterBottom variant="h5">
                        Weather Summary
                    </Typography>
                    <Typography variant="body2" >
                        Weather: {props.text}
                    </Typography>
                    <Typography variant="body2" >
                        Country: {props.country}
                    </Typography>
                    <Typography variant="body2" >
                        City: {props.name}
                    </Typography>
                    <Typography variant="body2" >
                        Local Time: {props.localtime}
                    </Typography>
                    <Typography variant="body2" >
                        Temperature: {props.temperature}
                    </Typography>
                </CardContent>
            </Card>

        </Box>

    )
}



