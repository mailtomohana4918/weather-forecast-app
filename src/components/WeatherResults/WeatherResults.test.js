import { render } from "@testing-library/react";
import React from "react";
import {WeatherResults} from "./WeatherResults";

describe("WeatherResults Component", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<WeatherResults {...{text:"sunny",country:"United Kingdom",name:"London",localtime:"",temperature:25}}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});