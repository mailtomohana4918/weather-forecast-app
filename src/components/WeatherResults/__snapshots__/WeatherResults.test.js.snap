// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherResults Component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 makeStyles-boxContainer-1"
  >
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      sx="[object Object]"
    >
      <img
        alt="weather"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
        height="140"
        src="weather.jpg"
      />
      <div
        class="MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Weather Summary
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Weather: sunny
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Country: United Kingdom
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          City: London
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Local Time: 
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Temperature: 25
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
